### PR TITLE
Cs/sc 3104 add mobile workers page gps capture page

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description

This PR alters the GPS capture page for the cases by adding a new tab for capturing mobile worker GPS data. In addition, when the "Capture on Map" button is clicked the map will appear (hidden by default) where the user can select the location of the case/user. Upon selecting a point on the map a marker appears which is draggable. The coordinates of the marker will update the relevant coordinate input fields (lat/lon) of the user/case. 

Also, a banner will show below the table listing the mobile workers indicating that new mobile workers can be created on the mobile workers page. A link to that page is also provided. 

### The new look of the _Manage GPS Data_ page

image

### The _Update Mobile Worker Data_ tab

image
   

### When the user selects "Capture on Map" and click somewhere

image

## Technical Summary
The _gps_capture.html_ page was altered to be reused as a component/partial and a new page `manage_gps_data.html` page was added which serves as the base of the _Manage GPS Data_ page.

Three global variables are [introduced](https://github.com/dimagi/commcare-hq/blob/8bf17cf3808a5dc8975d67b3feecd7d2c5873341/corehq/apps/geospatial/static/geospatial/js/gps_capture.js#L17-L19) to manage the selected item and it's marker on the map for which the GPS coordinates are being captured on the map.  

## Feature Flag
Geospatial feature

### Safety story
Tested this locally

### Automated test coverage
No tests included

### QA Plan
No formal QA planned at this stage.

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Charl1996** [comment]
Just a note, the markers are not bright green anymore, since I left it to be the default color.

File: corehq/apps/geospatial/static/geospatial/js/gps_capture.js [hunk 1]
```diff
@@ -197,25 +201,19 @@ hqDefine("geospatial/js/gps_capture",[
         return self;
     };
 
-    function updateSelectedItemLonLat(lon, lat) {
-        selectedDataListItem.setCoordinates(lat, lon);
+    function setMarkerAtLngLat(lon, lat) {
+        mapMarker.remove();
+        mapMarker.setLngLat([lon, lat]);
+        mapMarker.addTo(map);
```
[the remark this answers is not in the document]
Comment (Charl1996) [reply corehq/apps/geospatial/static/geospatial/js/gps_capture.js:207]: Yeah

File: corehq/apps/geospatial/static/geospatial/js/gps_capture.js [hunk 2]
```diff
@@ -11,6 +11,38 @@ hqDefine("geospatial/js/gps_capture",[
...
+            centerMapWithMarker();
+        }
+        else {
+            $("#" + MAP_CONTAINER_ID).hide();
+        }
+    };
+
+    function centerMapWithMarker() {
+        if (selectedDataListItem) {
+            let dataItem = selectedDataListItem.itemLocationBeingCapturedOnMap();
```
[the remark this answers is not in the document]
Comment (Charl1996) [reply corehq/apps/geospatial/static/geospatial/js/gps_capture.js:37]: The `selectedDataListItem` variable holds an instance of the `dataItemListModel` object which contains the `itemLocationBeingCapturedOnMap`, which, in turn, is the specific item of the `dataItemListModel` instance which is selected.

Maybe I should change the variable name from `selectedDataListItem` to simply `selectedDataListObject` or something. The "item" in the name might make this confusing.

[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/geospatial/static/geospatial/js/gps_capture.js:37]: oh its the list! Ya, must rename that.
Ya it was the `item` in the name that caused the confusion, `selectedDataList` would have been clear.

File: corehq/apps/geospatial/static/geospatial/js/gps_capture.js [hunk 3]
```diff
@@ -238,7 +243,16 @@ hqDefine("geospatial/js/gps_capture",[
...
+    function TabListViewModel() {
+        var self = {};
+        self.onclickAction = function() {
+            collapseMap();
+        }
+        return self;
+    }
+
     $(function () {
+        $("#tabs-list").koApplyBindings(TabListViewModel());
```
[the remark this answers is not in the document]
Comment (Charl1996) [reply corehq/apps/geospatial/static/geospatial/js/gps_capture.js:252]: @mkangia 
The problem is that the `dataItemListModel` binding is made deeper within the html and at this stage if we want to bring the tabs level logic into the `dataItemListModel` we'd have to make massive changes to the `dataItemListModel`. To capture the tabs selection I can do one of two things,

1. Create custom view model binding (as done here)
2. Use jQuery to fire function on click.

...unless there's some other way of doing this?

[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/geospatial/static/geospatial/js/gps_capture.js:252]: Just confirming that what I meant was that if we could add
```
self.onclickAction = function () {
    resetMap();
};
```
to dataItemListModel? I guess that would need you to move the "li" in the template as well. Fine to ignore this if that is going to mud things more than you would want to.

**Review by zandre-eng — APPROVED** [review]
Nothing blocking from my side.

**Review by mkangia — APPROVED** [review]

**Pull request merged**